### DISCUSSION TITLE
LGCP models not fitting

### DISCUSSION THREAD

**Discussion by naph0n** [inlabru-org/inlabru#180] [Q&A] [answered]
Hi,

I am currently trying to fit a LGCP model to aerial survey data, which is split into 27 months (one data set per month). I have managed to obtain results for 25 months, but for some reason the code does not work for 2 of these, and keeps returning this message on hundreds of lines:

`GMRFLib_2order_approx: rescue NAN/INF values in logl for idx=0`

Sometimes it keeps recycling this message and goes round in circles before I eventually stop it, and sometimes it just crashes and produces an error message saying it has crashed. 

I can't understand what is different with the data for these 2 months, especially as I managed to fit models for these data recently using the same code. 

I was wondering if anyone has had a similar error or if I am doing something wrong?

Thanks!

**Comment by finnlindgren** [comment, marked as answer]
The warning message usually indicates a model with problematic numerics, for one reason or other (e.g. badly scaled, incorrectly specified, etc). What inlabru and INLA versions are you running it on?  What spatial units are used for the mesh?

**Reply by naph0n** [reply]
I see, it's just strange that I was able to fit it before, but now isn't working at all! I'm using inlabru version 2.7.0 and INLA 22.05.07. The spatial units are in km

**Reply by finnlindgren** [reply]
There's been a lot of numerical developments in INLA in the past year, and this may be related to that. If you can't upgrade to the latest version (22.12.16, should work for Linux and Windows and some Macs, but for some Macs there seems to be a missing file that breaks it; I've alerted Haavard to that), you can try the "experimental" computation mode:
```
fit <- bru(..., options = list(..., inla.mode = "experimental"))
```
(That's the default from 22.12.16, and has a new name, "compact")
It may also be worth trying the inlabru development version (see https://inlabru-org.github.io/inlabru/ for installation options)

**Reply by naph0n** [reply]
ah that seems to have worked, thank you!
